tests/kokkos/kokkos_utils_tests.cpp

```diff
@@ -244,4 +245,81 @@ TEST_CASE("parallel_reduce", "[kokkos_utils]")
   test_parallel_reduce<Real,15,false> ();
 }
 
+
+template<typename Scalar, int TotalSize, int VectorSize, bool Serialize>
+void test_view_reduction()
+{
+  using Device = ekat::DefaultDevice;
+  using MemberType = typename ekat::KokkosTypes<Device>::MemberType;
+  using ExeSpace = typename ekat::KokkosTypes<Device>::ExeSpace;
+
+  ekat::ExeSpaceUtils<Serialize,ExeSpace> ExeSpaceUtils; 
+  
+  using PackType = ekat::Pack<Scalar, VectorSize>;
+  using ViewType = Kokkos::View<PackType*,ExeSpace>;
+
+  const int view_length = (int)std::ceil(1.0*TotalSize/VectorSize);
+
+  Scalar serial_result = Scalar();
+  ViewType data("data", view_length);
+  const auto data_h = Kokkos::create_mirror_view(data);
+  auto raw = data_h.data();
+  for (int k = 0; k < view_length; ++k) {
+    for (int p=0; p<VectorSize; ++p) {
+      if (k*VectorSize+p >= TotalSize) {
+        // represents pack garbage
+        raw[k][p] = 1e100;
```

Comment:
You could use `ScalarTraits<Scalar>::invalid()`, which sets garbage to quiet nans.